dom.go

```diff
@@ -258,6 +262,7 @@ func (h *HTML) Restore(old ComponentOrHTML) {
 	for name, value := range h.dataset {
 		dataset.Set(name, value)
 	}
+	h.populateClassNameProperty()
```

Comment:
In this situation, the behavior should just be inlined (this is element creation, after all)